core/src/test/scala/org/apache/spark/sql/delta/DeltaSuite.scala

```diff
@@ -1647,5 +1647,24 @@ class DeltaSuite extends QueryTest
     dir2.renameTo(dir1)
     checkAnswer(spark.read.format("delta").load(dir1.getCanonicalPath), spark.range(10).toDF)
   }
+
+  test("set metadata upon write") {
+    withTempDir { inputDir =>
+      val testPath = inputDir.getCanonicalPath
+      spark.range(10)
+        .map(_.toInt)
+        .withColumn("part", $"value" % 2)
+        .write
+        .format("delta")
+        .option("delta.sampleRetentionDuration", "123 days")
```

Comment:
nit: could you add an non delta option here so that we can verify we don't write non delta options to metadata?